pandas/core/indexes/base.py

```diff
@@ -3049,8 +3049,9 @@ def _get_nearest_indexer(self, target: "Index", limit, tolerance) -> np.ndarray:
         left_indexer = self.get_indexer(target, "pad", limit=limit)
         right_indexer = self.get_indexer(target, "backfill", limit=limit)
 
-        left_distances = np.abs(self[left_indexer] - target)
-        right_distances = np.abs(self[right_indexer] - target)
+        target = np.asarray(target)
+        left_distances = np.abs(self.values[left_indexer] - target)
+        right_distances = np.abs(self.values[right_indexer] - target)
```

Comment:
what happens if instead of casting to ndarray this is `self._values[indexer] - target._values`?  That will avoid troublesome casting for EAs.

Reply:
and if i understand correctly, make the overriding in datetimelike unnecessary

Reply:
Oh cool, that works!